src/ConnectedMode/Migration/MigrationChecker.cs

```diff
@@ -0,0 +1,78 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Binding;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Migration
+{
+    [Export(typeof(MigrationChecker))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class MigrationChecker : IDisposable
+    {
+        private readonly IActiveSolutionTracker activeSolutionTracker;
+        private readonly IGoldBarController goldBarController;
+        private readonly IConfigurationProvider configurationProvider;
+        private readonly IObsoleteConfigurationProvider obsoleteConfigurationProvider;
+
+        [ImportingConstructor]
+        public MigrationChecker(
+            IActiveSolutionTracker activeSolutionTracker,
+            IGoldBarController goldBarController,
+            IConfigurationProvider configurationProvider,
+            IObsoleteConfigurationProvider obsoleteConfigurationProvider)
+        {
+            this.activeSolutionTracker = activeSolutionTracker;
+            this.goldBarController = goldBarController;
+            this.configurationProvider = configurationProvider;
+            this.obsoleteConfigurationProvider = obsoleteConfigurationProvider;
+
+            activeSolutionTracker.ActiveSolutionChanged += OnActiveSolutionChanged;
+
+            // Initial check incase event was fired before we registered.
+            DisplayGoldBarIfMigrationIsNeeded();
+        }
+
+        private void OnActiveSolutionChanged(object sender, ActiveSolutionChangedEventArgs args)
+        {
+            if (args.IsSolutionOpen)
```

Comment:
Do we need to close the goldbar if the solution closes?

Reply:
Yes, I think we should.